interface-definitions/interfaces_thunderbolt.xml.in

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="interfaces">
+    <children>
+      <tagNode name="thunderbolt" owner="${vyos_conf_scripts_dir}/interfaces_thunderbolt.py">
+        <properties>
+          <help>Thunderbolt Interface</help>
+          <priority>330</priority>
+          <constraint>
+            <regex>thunderbolt[0-9]+</regex>
+          </constraint>
+          <constraintErrorMessage>Thunderbolt interface must be named thunderboltN</constraintErrorMessage>
+          <valueHelp>
+            <format>thunderboltN</format>
+            <description>thunderbolt interface name</description>
+          </valueHelp>
+        </properties>
+        <children>
+          #include <include/interface/address-ipv4-ipv6-dhcp.xml.i>
+          #include <include/generic-description.xml.i>
+          #include <include/interface/dhcp-options.xml.i>
+          #include <include/interface/dhcpv6-options.xml.i>
+          #include <include/interface/disable-link-detect.xml.i>
+          #include <include/interface/disable.xml.i>
+          #include <include/interface/vrf.xml.i>
+          #include <include/interface/mirror.xml.i>
+          #include <include/interface/offload.xml.i>
+          #include <include/interface/ipv4-options.xml.i>
+          #include <include/interface/ipv6-options.xml.i>
+          #include <include/interface/mac.xml.i>
+          #include <include/interface/mtu-68-16000.xml.i>
+          <leafNode name="mtu">
+            <defaultValue>1500</defaultValue>
+          </leafNode>
+          #include <include/interface/redirect.xml.i>
+          #include <include/interface/vif-s.xml.i>
+          #include <include/interface/vif.xml.i>
+          <leafNode name="pci-address">
+            <properties>
+              <help>PCI device address of the Thunderbolt controller port</help>
+              <completionHelp>
+                <list>00:0d.2 00:0d.3</list>
```

Comment:
I think those addresses are dynamic from host to host. It should be a completion helper script (best BASH to avoid process startup penalty)

Reply:
Any progress on this?